plugins/services/src/js/data/MesosClient.ts

```diff
@@ -0,0 +1,75 @@
+//@ts-ignore
+import { request } from "@dcos/mesos-client";
+import { Observable } from "rxjs";
+import { map } from "rxjs/operators";
+import { QuotaData, quotaFields } from "../types/Quota";
+import ValidatorUtil from "#SRC/js/utils/ValidatorUtil";
+
+interface QuotaRequestValue {
+  value: number;
+}
+
+interface QuotaConfig {
+  role: string;
+  guarantees?: Record<string, QuotaRequestValue>;
+  limits?: Record<string, QuotaRequestValue>;
+}
+
+interface UpdateQuota {
+  force?: boolean;
+  quota_configs: QuotaConfig[];
+}
+
+interface UpdateQuotaRequest {
+  type: "UPDATE_QUOTA";
+  update_quota: UpdateQuota;
+}
+
+function roleFromGroupId(groupId: string): string {
+  return groupId.split("/").pop() || groupId;
+}
+
+function limitsFromQuotaFormData(
+  quotaData: QuotaData
+): Record<string, QuotaRequestValue> {
+  const result: Record<string, QuotaRequestValue> = {};
+  for (let field of quotaFields) {
+    const value = quotaData[field].trim();
+    if (!ValidatorUtil.isEmpty(value)) {
+      result[field] = {
+        value: parseFloat(value)
+      };
+    }
+  }
+  return result;
+}
+
+function makeUpdateQuota(groupId: string, quotaData: QuotaData): UpdateQuota {
+  return {
+    force: quotaData.force,
+    quota_configs: [
+      {
+        role: roleFromGroupId(groupId),
+        limits: limitsFromQuotaFormData(quotaData)
+      }
+    ]
+  };
+}
+
+export function updateQuota(
```

Comment:
I also added some mesos request wrappers when building maintenance features, but I added them as some simple actions-- do you think this should eventually define the entire API client? The name hints at that, but it seems overly ambitious. Maybe this should just be the QuotaActions instead?

Reply:
My idea was this could be abstracted out to a package and be a more robust client.

For plans I wrote a very specific `SDKPlansClient` and that works, but I assumed as we added more Mesos calls to data-layer having a central client would be useful, same with `MarathonClient`. But that certainly could be too ambitious.

Reply:
Maybe I should of put these in `packages/` to begin with.